src/modules/ims_isc/ims_isc_mod.c

```diff
@@ -244,10 +244,17 @@ int isc_match_filter(struct sip_msg *msg, char *str1, udomain_t *d)
 	int free_s = 0;
 
 	//the callback from the Cx interface in case of unreg terminating initial message is a FAILURE_ROUTE. Hence we need an addl. flag
+	// meaning of firstflag:
+	//    firstflag == 0:
+	//       request route: request has been received from AS, old ISCMARK found in Route header field
+	//       failure route: request to AS has been rejected or timed out, old ISCMARK found in lumps
```

Comment:
Does it makes sense to have 4 possible values instead of just 2? Is that even possible though? I'm thinking if that would make things more clear, instead of having to remember that each of the current values have 2 sub-cases.